### ISSUE TITLE
Acurite to Home Assistant Erroring out after a few minutes

### ISSUE THREAD

**Issue by hartman44** [thejeffreystone/hassio_addons#33]
I am getting strange behavior where this will work for about two minutes and then error out. If I restart the add-on it starts working again for a few minutes and then erroring out again.

My setup it is running HA on a Virtual Box on a Windows 11 machine. I am also using my RTL Dongle with rtlamr2mqtt. Not sure if this is causing a conflict. Would like some insight on whether I can get this running reliably, thanks.

Config:

mqtt_host: XXXX
mqtt_port: 1883
mqtt_user: XXXX
mqtt_password: XXXX
mqtt_topic: rtl_433
mqtt_retain: 'true'
protocol: '-R 11 -R 40 -R 41 -R 55 -R 74'
whitelist_enable: true
whitelist: '11877'
expire_after: 0
units: customary
discovery_prefix: homeassistant
discovery_interval: 600
debug: false

> Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:10-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98576, "rssi": -0.138432, "snr": 20.86633, "noise": -21.0048}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:27-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98682, "rssi": -0.134052, "snr": 18.06563, "noise": -18.1997}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:27-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98682, "rssi": -0.134052, "snr": 18.06563, "noise": -18.1997}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:27-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98682, "rssi": -0.134052, "snr": 18.06563, "noise": -18.1997}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:43-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98848, "rssi": -0.105988, "snr": 23.23008, "noise": -23.3361}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:43-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98848, "rssi": -0.105988, "snr": 23.23008, "noise": -23.3361}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
Received message: rtl_433/events : {"time": "2022-06-24 13:26:43-0400", "protocol": 40, "model": "Acurite-Tower", "id": 11877, "channel": "A", "battery_ok": 1, "temperature_F": 85.28, "humidity": 51, "mic": "CHECKSUM", "mod": "ASK", "freq": 433.98848, "rssi": -0.105988, "snr": 23.23008, "noise": -23.3361}
Received key time for Acurite-Tower
Received key channel for Acurite-Tower
Received key battery_ok for Acurite-Tower
Received key temperature_F for Acurite-Tower
Received key humidity for Acurite-Tower
Received key freq for Acurite-Tower
Received key rssi for Acurite-Tower
Received key snr for Acurite-Tower
Received key noise for Acurite-Tower
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
LIBUSB_ERROR_NOT_FOUND: Entity not found!
Check your RTL-SDR dongle, USB cables, and power supply.
WARNING: async read failed (-5).
Library error -5, exiting...

>

**Comment by thejeffreystone** [comment]
Do you have the USB dongle in an extension cable? I think plugging it in directly sometimes causes issues like this.

**Comment by hartman44** [comment]
Yes, I have had it plugged into an extension USB cable from day one.

**Comment by thejeffreystone** [comment]
So after doing some digging this error has to do with the hardware. Something happening with the connection between the device and the usb stick. 

I will have to see if there is a way for me to catch these and then auto restart the add-on. But unfortunately my solution is only going to be trying to recover from this state since I cannot control or see the connection between the stick and the device.
